client.go

```diff
@@ -37,33 +44,72 @@ func NewClient(c Config) (*Client, error) {
 	}
 
 	return &Client{
-		client: goredis.NewClient(&goredis.Options{
-			Addr:     c.Addr,
+		client:  RedisClient{client:redis.NewClient(&redis.Options{
+			Addr: c.Addr,
 			Password: c.Password,
-			DB:       c.Database,
-		}),
-		ttl: c.TTL,
+			DB: c.Database,
+		})},
+		ttl:     0,
 	}, nil
 }
 
-//Set add session to redis
+// Set - add session to redis
 func (c *Client) Set(s *session.Session) error {
 	if s == nil {
 		log.Event(ctx, "session is empty", log.ERROR)
 		return errors.New("session is empty")
 	}
 
-	json, err := s.MarshalJSON()
+	sJSON, err := s.MarshalJSON()
 	if err != nil {
 		log.Event(ctx, "failed to marshal session", log.Error(err), log.ERROR)
 		return err
 	}
 
-	msg, err := c.client.Set(s.ID, json, c.ttl).Result()
+	msg, err := c.client.Set(s.ID, sJSON, c.ttl).Result()
 	if err != nil {
 		log.Event(ctx, msg, log.Error(err), log.ERROR)
+		return err
 	}
-	log.Event(ctx, msg, log.INFO)
 
 	return nil
 }
+
+// GetByID - gets a session from redis using its ID
+func (c *Client) GetByID(id string) (*session.Session, error) {
+	if id == "" {
+		log.Event(ctx, "id value is blank", log.ERROR)
+		return nil, errors.New("id value is blank")
```

Comment:
Consider making this error a public const - it allows the caller to "switch" on the error type if they need to. Also consider making the message more descriptive for example `session id required by was empty` gives you a much clearer idea of the error without having to dig into the code.

Reply:
Made a few of the error messages constants.